src/components/PatientRecord/PatientRecord.tsx

```diff
@@ -127,7 +132,9 @@ const PatientRecordElement: FC<PatientRecordElementProps> = ({ resourceType }) =
 
 const Visualizer: FC<VisualizerProps> = ({ resourceType, resourcesByType }) => {
   const patient = usePatient().patient as fhir.Patient;
-  if (resourceType === 'Patient') return <PatientVisualizer patient={patient} />;
+
+  if (resourceType === 'Pathway') return <PathwayVisualizer />;
```

Comment:
So the first time I saw this I thought a `switch` would be appropriate, but now I have another thought. We're creating a list of resource types above and then iterating over it and using this big if to pick elements out later. How would you feel about instead having it laid out like this in the PatientRecord:

```
<PatientRecordElement resourceType="Patient">
  <PatientVisualizer patient={patient} />
</PatientRecordElement>
<PatientRecordElement resourceType="Condition">
  <ConditionsVisualizer rows={getResourcesByType("Condition"} />
</PatientRecordElement>
...
```
(with variables re-oriented as needed)
In other words, the PatientRecordElement wouldn't know which visualizer to render, it would just be passed in as a child element to be rendered.

(I grant that ultimately it doesn't change much, I just don't like this giant list of ifs 😛 )

Reply:
One of the downsides of this is that `PatientRecordElement` already calls `getResourcesByType` so this would do the search twice. It already bothers me this is r*O(n) where r is the number of resource types and n is the number of resources in the patient record (ideally we would make it O(n)). 
I can't think of a _clean_ (or at least cleaner than the if block) way to share the result of `getResourcesByType` in this structure.

Reply:
We can use a group-by function to make it O(n), lodash has one and I think we include lodash indirectly as a dependency of something else, or see https://stackoverflow.com/questions/14446511/most-efficient-method-to-groupby-on-an-array-of-objects

Reply:
So I implemented the long list and the group by function but honestly I think it is 10 times cleaner using the big if statement. The more I write the more I'm itching to change it back because it looks harder to read IMO
```
<PatientRecordElement
       resourceType={'Condition'}
       resourceCount={getResourceByType('Condition', groupedResources).length}
>
            <ConditionsVisualizer rows={getResourceByType('Condition', groupedResources)} />
</PatientRecordElement>
 <PatientRecordElement
       resourceType={'Observation'}
       resourceCount={getResourceByType('Observation', groupedResources).length}
>
            <ObservationsVisualizer rows={getResourceByType('Observation', groupedResources)} />
</PatientRecordElement>
....
```

Reply:
The reason it looks so bad is that the group by function returns a map so doing `groupedResources.get(resourceType)` returns `DomainResource[] | undefined`.  This is unavoidable but I don't like the fact we have to keep repeating the resource type over and over again